### PR TITLE
BUG Fix SiteTree / SiteConfig permissions (3.0 backport version)

### PR DESCRIPTION
**Pull request by tractorcow**

### CODE REVIEW

File: code/model/SiteConfig.php [hunk 1]
```diff
@@ -192,48 +205,70 @@ static public function make_site_config() {
 	 * called if a page is set to Inherit, but there is nothing
 	 * to inherit from.
 	 *
-	 * @param mixed $member 
+	 * @param Member $member
 	 * @return boolean
 	 */
-	public function canView($member = null) {
```
Comment (hafriedlander) [inline code/model/SiteConfig.php]: You remove canView here, but don't later replace it. Is the default from DataObject enough? (You do replace canEdit later)

File: code/controllers/CMSPageAddController.php [hunk 2]
```diff
@@ -21,7 +21,7 @@ public function AddForm() {
 		$pageTypes = array();
 		foreach($this->PageTypes() as $type) {
 			$html = sprintf('<span class="page-icon class-%s"></span><strong class="title">%s</strong><span class="description">%s</span>',
-				$type->getField('Title'),
+				$type->getField('ClassName'),
```
Comment (hafriedlander) [inline code/controllers/CMSPageAddController.php]: What's the purpose of this change? Doesn't seem related to the rest of the patch?

Comment (tractorcow) [inline code/controllers/CMSPageAddController.php]: It was an error which the later fixes to the .js relies on. ;)

File: code/model/SiteConfig.php (hunk 1 above)
Comment (tractorcow) [inline code/model/SiteConfig.php]: canView has been reinstated with a logical replacement.
